pkg/cluster/ensureendpoints.go

```diff
@@ -36,3 +64,49 @@ func (m *manager) getSubnetIds() ([]string, error) {
 	}
 	return subnets, nil
 }
+
+// addEndpointsToSubnet adds the endpoints (that either are missing in subnet
+// or aren't in succeeded state in the subnet) to the subnet and returns the updated subnet
+func addEndpointsToSubnet(endpoints []string, subnet *armnetwork.Subnet) (subnetChanged bool) {
+	for _, endpoint := range endpoints {
+		endpointFound, serviceEndpointPtr := subnetContainsEndpoint(subnet, endpoint)
+
+		if !endpointFound || *serviceEndpointPtr.ProvisioningState != armnetwork.ProvisioningStateSucceeded {
```

Comment:
Can ProvisioningState ever be in an error state? We wouldn't want to add it then I'm guessing

Reply:
This is a copy/paste from the old code.
Also this function adds endpoints that are not available in the subnet, so it's supposed to be ok.